### PR TITLE
Tls ingress

### PR DESCRIPTION
**Pull request by danielezer**
Add an option to specify the certificate and the private key for the ingress tls

### CODE REVIEW

File: charts/my-bloody-jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -61,3 +61,7 @@ Define default values
 {{- define "my-bloody-jenkins.persistentVolumeClaimName" -}}
 {{- .Values.persistenceExistingClaim | default (include "my-bloody-jenkins.fullname" .) -}}
 {{- end -}}
+
+{{- define "my-bloody-jenkins.tlsSecretName" -}}
+{{- printf "%s-tls-secret" (include "my-bloody-jenkins.fullname" .) -}}
+{{- end -}}
```
Comment (odavid) [inline charts/my-bloody-jenkins/templates/_helpers.tpl:67]: Please add a newline here.

File: charts/my-bloody-jenkins/templates/ingress-tls-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    app: {{ template "my-bloody-jenkins.name" . }}
+    chart: {{ template "my-bloody-jenkins.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+type: kubernetes.io/tls
+data:
+  tls.crt: {{ .Values.ingress.tls.certificate | b64enc }}
+  tls.key: {{ .Values.ingress.tls.privateKey | b64enc }}
+{{- end }}
```
Comment (odavid) [inline charts/my-bloody-jenkins/templates/ingress-tls-secret.yaml:15]: Same goes here.

File: charts/my-bloody-jenkins/README.md [hunk 3]
```diff
@@ -137,6 +137,8 @@ The following table lists the configurable parameters of the chart and their def
 | `ingress.annotations`        | Ingress Annoations| `{}`
 | `ingress.hostname`        | Ingress Hostname (Required only if ingress is enabled)|
 | `ingress.tls.secretName`        | Ingress TLS Secret Name - if provided, the ingress will terminate TLS|
```
Comment (odavid) [inline charts/my-bloody-jenkins/README.md]: I believe the documentation should state that secretName is mutual exclusive to certificate and privateKey.

**Review by odavid — CHANGES_REQUESTED** [review]
Hi @danielezer - Thanks for the PR! (: 
Please also increase the Chart version, so when merged it will be released automatically.

File: charts/my-bloody-jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (danielezer) [reply charts/my-bloody-jenkins/templates/_helpers.tpl:67]: Not sure I understand what you mean. Add a new line where?

**Comment by danielezer** [comment]
@odavid Why is the travis build failing? is this expected?

File: charts/my-bloody-jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (odavid) [reply charts/my-bloody-jenkins/templates/_helpers.tpl:67]: Newlines should be added to the end of text files.
https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

**Comment by odavid** [comment]
> @odavid Why is the travis build failing? is this expected?

The build was not designed for PR... I will fix that soon.

**Review by odavid — APPROVED** [review]

File: charts/my-bloody-jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (danielezer) [reply charts/my-bloody-jenkins/templates/_helpers.tpl:67]: Cool. wasn't aware of that. thanks

**Pull request merged**